packages/rocketchat-issuelinks/client.js

```diff
@@ -5,10 +5,33 @@ import s from 'underscore.string';
 // @param {Object} message - The message object
 //
 
+const regexpDefault = /(?:^|\s|\n)(#[0-9]+)\b/g;
+let regexp = regexpDefault;
+
+Tracker.autorun(function() {
+	regexp = regexpDefault;
+
+	if (!RocketChat.settings.get('IssueLinks_Enabled')) {
+		return;
+	}
+
+	const format = RocketChat.settings.get('IssueLinks_Format');
+	if (!format) {
+		return;
+	}
+
+	try {
+		regexp = new RegExp(`(?:^|\\s|\\n)(#${ format })\\b`, 'g');
+	} catch (error) {
+		console.log(`"${ format }" is a bad format to regexp [rocketchat-issuelinks]`);
+	}
+});
+
 function IssueLink(message) {
 	if (s.trim(message.html) && RocketChat.settings.get('IssueLinks_Enabled')) {
-		message.html = message.html.replace(/(?:^|\s|\n)(#[0-9]+)\b/g, function(match, issueNumber) {
-			const url = RocketChat.settings.get('IssueLinks_Template').replace('%s', issueNumber.substring(1));
+		const template = RocketChat.settings.get('IssueLinks_Template');
+		message.html = message.html.replace(regexp, function(match, issueNumber) {
+			const url = template.replace('%s', issueNumber.substring(1));
```

Comment:
IMO we should not do `.substring(1)` because we cannot be sure the regex will always have a single identifier character on begin.. to do this we could just change the regex to `#(${ format })` instead of `(#${ format })`